extension/agenthealth/handler/stats/provider/statuscode.go

```diff
@@ -0,0 +1,180 @@
+	statusCodeStatsOnce.Do(func() {
+		handler := &StatusCodeHandler{}
+		handler.filter = filter
+		handler.startResetTimer()
+		statusCodeSingleton = handler
+	})
+	return statusCodeSingleton
+}
+
+// startResetTimer initializes a reset timer to clear stats every 5 minutes.
+func (h *StatusCodeHandler) startResetTimer() {
+	h.resetTimer = time.AfterFunc(statusResetInterval, func() {
+		h.mu.Lock()
+		defer h.mu.Unlock()
+
+		h.statsByOperation.Range(func(key, _ interface{}) bool {
+			h.statsByOperation.Delete(key)
+			return true
+		})
+		log.Println("Status code stats reset.")
+		h.startResetTimer()
+	})
+}
+
+// HandleRequest is a no-op for the StatusCodeHandler.
+func (h *StatusCodeHandler) HandleRequest(ctx context.Context, _ *http.Request) {}
+
+// HandleResponse processes the HTTP response to update status code stats.
+func (h *StatusCodeHandler) HandleResponse(ctx context.Context, r *http.Response) {
+	// Extract the operation name
+	operation := awsmiddleware.GetOperationName(ctx)
+	if operation == "" {
+		log.Println("No operation name found in the context")
+		return
+	} else if !h.filter.IsAllowed(operation) {
+		log.Printf("Operation %s is not allowed", operation)
+		return
+	} else {
+		log.Printf("Processing response for operation: %s", operation)
+	}
+
+	operation = GetShortOperationName(operation)
+	statusCode := r.StatusCode
+
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	value, loaded := h.statsByOperation.LoadOrStore(operation, &[5]int{})
+	if !loaded {
+		log.Printf("Initializing stats for operation: %s", operation)
+	}
+	stats := value.(*[5]int)
+
+	h.updateStatusCodeCount(stats, statusCode, operation)
+
+	h.statsByOperation.Store(operation, stats)
+
+	h.statsByOperation.Range(func(key, value interface{}) bool {
+
+		operation := key.(string)
+		stats := value.(*[5]int)
+		log.Printf("Operation: %s, 200=%d, 400=%d, 408=%d, 413=%d, 429=%d", operation, stats[0], stats[1], stats[2], stats[3], stats[4])
+		return true
+	})
+}
+
+// Helper function to update the status code counts
+func (h *StatusCodeHandler) updateStatusCodeCount(stats *[5]int, statusCode int, operation string) {
+	switch statusCode {
+	case 200:
+		stats[0]++
+	case 400:
+		stats[1]++
+	case 408:
+		stats[2]++
+	case 413:
+		stats[3]++
+	case 429:
+		stats[4]++
+	default:
+		log.Printf("Received an untracked status code %d for operation: %s", statusCode, operation)
+	}
+}
+
+func GetShortOperationName(operation string) string {
+	switch operation {
+	case "PutMetricData":
+		return "pmd"
+	case "DescribeInstances":
+		return "di"
+	case "DescribeTags":
+		return "dt"
+	case "DescribeVolumes":
+		return "dv"
+	case "DescribeContainerInstances":
+		return "dci"
+	case "DescribeServices":
+		return "ds"
+	case "DescribeTaskDefinition":
+		return "dtd"
+	case "ListServices":
+		return "ls"
+	case "ListTasks":
+		return "lt"
+	case "CreateLogGroup":
+		return "clg"
+	case "CreateLogStream":
+		return "cls"
+	case "AssumeRole":
+		return "sts"
```

Comment:
I'd recommend following your convention and stick to an abbreviation of the operation rather than use the service (so `ar` instead of `sts`). It would make adding another sts service call confusing otherwise.